Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -121,6 +123,19 @@ - (void)setupWith:(SentryReplayOptions *)replayOptions
     }
 
     _notificationCenter = SentryDependencyContainer.sharedInstance.notificationCenterWrapper;
+    _dateProvider = SentryDependencyContainer.sharedInstance.dateProvider;
+
+    // The asset worker queue is used to work on video and frames data.
+    // Use a relative priority of -1 to make it lower than the default background priority.
+    _replayAssetWorkerQueue = [SentryDispatchQueueWrapper
+        createBackgroundDispatchQueueWithName:"io.sentry.session-replay.asset-worker"
+                             relativePriority:-1];
+    // The dispatch queue is used to asynchronously wait for the asset worker queue to finish its
+    // work. To avoid a deadlock, the priority of the processing queue must be lower than the asset
+    // worker queue. Use a relative priority of -2 to make it lower than the asset worker queue.
+    _replayProcessingQueue = [SentryDispatchQueueWrapper
+        createBackgroundDispatchQueueWithName:"io.sentry.session-replay.processing"
```

Comment:
`h`: The comment mentions that setting the wrong priorities here can lead to a deadlock. I think it's essential to have a test fail when using the wrong priorities to avoid regressions. We could assert that the `SentryDispatchQueueWrapper` is initialized with the correct attributes, which tests implementation details, and has the downside of having to change the test when the implementation details change. We could also have a test causing a deadlock when interacting with this class when the priorities are wrongly set, which could be a bit harder to achieve, but has the advantage of not testing implementation details.

Reply:
See my answer to the other comments on how I'll make sure that SentryDispatchQueueWrapper is called correctly.

I am not sure yet if it's viable to create a failing test if the priorities are wrong, as we are doing this because the inner workings of AVFoundation are not transparent, but Xcode shows a warning.

Reply:
As mentioned in the other PR comment, I added tests